scripts/pylib/twister/twisterlib/platform.py

```diff
@@ -71,9 +71,28 @@ def load(self, platform_file):
         self.simulation = data.get('simulation', "na")
         self.simulation_exec = data.get('simulation_exec')
         self.supported_toolchains = data.get("toolchain", [])
-        if self.arch == 'arm':
-            if 'armclang' not in self.supported_toolchains:
-                self.supported_toolchains.append('armclang')
+        if self.supported_toolchains == None:
+            self.supported_toolchains = []
+
+        support_toolchain_variants = {
+          "arc": ["zephyr", "cross-compile", "xtools", "arcmwdt"],
+          "arm": ["zephyr", "gnuarmemb", "xtools", "armclang", "llvm"],
+          "arm64": ["zephyr", "cross-compile"],
+          "mips": ["zephyr", "xtools"],
```

Comment:
do we still do xtools? I thought we had plans to drop this, @stephanosio

Reply:
We deprecated it in 3.3. I put it in here for now, we can remove whenever we want.

Reply:
ok, I like the map better now, resembles what I was talking about in thr Arch meeting.